static/js/admin.js

```diff
@@ -408,6 +428,67 @@ exports.setup_page = function () {
             }
         });
     });
+
+    $('.admin_emoji_table').on('click', '.delete', function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var btn = $(this);
+
+        channel.del({
+            url: '/json/realm/emoji/' + encodeURIComponent(btn.attr('data-emoji-name')),
+            error: function (xhr, error_type) {
+                if (xhr.status.toString().charAt(0) === "4") {
+                    btn.closest("td").html(
+                        $("<p>").addClass("text-error").text($.parseJSON(xhr.responseText).msg)
+                    );
+                } else {
+                     btn.text("Failed!");
```

Comment:
It would probably make sense, but I think this is the "canonical" way to do it (at least looking at admin.js). It sounds reasonable to me as we denote the table row within which the error occurs. I could change it if you think it makes sense, but I guess it would be a good idea to change the other few occurrences of this pattern in admin.js too.